### PR TITLE
implement directing appliances, add directed floodlight appliance

### PR DESCRIPTION
**Pull request by waxcatape**
#### Summary
Features "Implement setting direction for appliances and add a directed floodlight appliance"

#### Purpose of change

Allow for directed appliance light sources.

#### Describe the solution

Previously appliances didn't have light sources that illuminate a cone and allow for choosing a direction. This PR makes the logic already used by vehicles available to appliances and adds a directed floodlight appliance that illuminates a semicircle in a chosen direction. Upon placing an appliance that has a specific flag (for now only HALF_CIRCLE_LIGHT) a popup appears prompting the player to choose the direction of illumination, just like with car headlights. You can drag the appliance around to change the direction after placement.

#### Describe alternatives you've considered

Implement a more general furniture solution that doesn't leverage the vehicle code behind appliances. I'm not good enough. 

#### Testing

Spawned the new directed floodlight ingame, placed it using the construction menu, made sure that the prompt for setting the lighting direction is working, dragged the appliance around to see if lighting direction changes. 

The test suite came up green.

#### Additional context

This will be followed up by wall-mounted light fixtures using the new code in the near future.

### CODE REVIEW

File: src/construction.cpp [hunk 1]
```diff
@@ -1087,6 +1087,7 @@ void complete_construction( Character *you )
             debugmsg( "No space to displace items from construction finishing" );
         }
     }
+
```
Comment (irwiss) [inline src/construction.cpp:1090]: some whitespace changes still left over
```suggestion
```

File: src/veh_appliance.cpp [hunk 2]
```diff
@@ -61,9 +64,10 @@ void place_appliance( const tripoint &p, const vpart_id &vpart, const std::optio
 
     veh->add_tag( flag_APPLIANCE );
 
+    int partnum = -1;
     if( base ) {
         item copied = *base;
-        veh->install_part( point_zero, vpart, std::move( copied ) );
+        partnum = veh->install_part( point_zero, vpart, std::move( copied ) );
```
Comment (irwiss) [inline src/veh_appliance.cpp:70]: the else below should also set partnum for consistency

File: src/construction.cpp [hunk 3]
```diff
@@ -1496,6 +1497,8 @@ void construct::done_appliance( const tripoint_bub_ms &p, Character & )
     const item &base = components.front();
     const vpart_id &vpart = vpart_appliance_from_item( base.typeId() );
 
+    // Make some lighting appliances directed
+
```
Comment (irwiss) [inline src/construction.cpp:1501]:
```suggestion
```

File: src/veh_appliance.cpp (hunk 2 above)
Comment (waxcatape) [reply src/veh_appliance.cpp:70]: followed the suggestion

File: data/json/furniture_and_terrain/appliances.json [hunk 4]
```diff
@@ -632,6 +632,16 @@
...
   },
+  {
+    "id": "ap_standing_directed_floodlight",
+    "type": "vehicle_part",
+    "name": { "str": "standing directed floodlight" },
+    "item": "standing_directed_floodlight",
+    "description": "A large, heavy light that illuminates a wide semicircle around it when switched on.",
+    "copy-from": "directed_floodlight",
+    "flags": [ "HALF_CIRCLE_LIGHT", "ENABLED_DRAINS_EPOWER", "APPLIANCE", "CTRL_ELECTRONIC" ],
+    "requirements": { "removal": { "time": "6 m" } }
```
Comment (Fris0uman) [inline data/json/furniture_and_terrain/appliances.json:643]: It takes 5s to place down but 6m to pick back up?

**Pull request merged**